.github/workflows/main.yml

```diff
@@ -0,0 +1,29 @@
+name: main
+
+on: [push]
```

Comment:
do we want this to run on any push? or just a specific branch?

Reply:
we might want
```yml
on:
  push:
    branches:
      - main
```

Reply:
oh wait up, do we want to attempt to build on any push but publish only on pushes to main?

Reply:
Nah cause we can put a test command into that build step (#11), and the publish step only runs on main